src/components/MessageList/MessageList.tsx

```diff
@@ -55,85 +179,117 @@ const ErrorNotification = styled.View`
+      ? undefined
+      : message.id;
+  });
   const [newMessagesNotification, setNewMessageNotification] = useState(false);
 
   useEffect(() => {
-    const currentLastMessage = getLastReceivedMessage(messageList);
-    const currentLastReceivedId = currentLastMessage && currentLastMessage.id;
-    if (currentLastReceivedId) {
-      const hasNewMessage = lastReceivedId !== currentLastReceivedId;
-      const userScrolledUp = yOffset.current > 0;
-      const isOwner = currentLastMessage.user.id === client.userID;
+    const currentLastMessage = getLastReceivedMessage<
+      At,
+      Ch,
+      Co,
+      Ev,
+      Me,
+      Re,
+      Us
+    >(messageList);
+    if (
+      currentLastMessage &&
+      !isDateSeparator<At, Ch, Co, Ev, Me, Re, Us>(currentLastMessage)
+    ) {
+      const currentLastReceivedId = currentLastMessage.id;
+      if (currentLastReceivedId) {
+        const hasNewMessage = lastReceivedId !== currentLastReceivedId;
+        const userScrolledUp = yOffset.current > 0;
+        const isOwner = currentLastMessage?.user?.id === client.userID;
 
-      // always scroll down when it's your own message that you added..
-      const scrollToBottom =
-        (hasNewMessage && isOwner) || (hasNewMessage && !userScrolledUp);
+        // always scroll down when it's your own message that you added..
+        const scrollToBottom =
+          (hasNewMessage && isOwner) || (hasNewMessage && !userScrolledUp);
 
-      // Check the scroll position... if you're scrolled up show a little notification
-      if (!scrollToBottom && hasNewMessage && !newMessagesNotification) {
-        setNewMessageNotification(true);
-      }
-      if (scrollToBottom) {
-        if (flatListRef.current) {
-          flatListRef.current.scrollToIndex({ index: 0 });
+        // Check the scroll position... if you're scrolled up show a little notification
+        if (!scrollToBottom && hasNewMessage && !newMessagesNotification) {
+          setNewMessageNotification(true);
+        }
+        if (scrollToBottom) {
+          if (flatListRef.current) {
+            flatListRef.current.scrollToIndex({ index: 0 });
+          }
         }
-      }
 
-      // remove the scroll notification if we already scrolled down...
-      if (scrollToBottom && newMessagesNotification) {
-        setNewMessageNotification(false);
+        // remove the scroll notification if we already scrolled down...
+        if (scrollToBottom && newMessagesNotification) {
+          setNewMessageNotification(false);
+        }
+        if (hasNewMessage) setLastReceivedId(currentLastReceivedId);
       }
-      if (hasNewMessage) setLastReceivedId(currentLastReceivedId);
     }
   }, [messageList]);
 
   const loadMore = threadList ? loadMoreThread : mainLoadMore;
 
-  const renderItem = ({ item: message }) => {
-    if (message.type === 'message.date') {
+  const renderItem = (message: InsertDate<At, Ch, Co, Ev, Me, Re, Us>) => {
```

Comment:
hmm this might be an issue since Message component is expecting...
`message: MessageWithDates<At, Ch, Co, Me, Re, Us>;`

Reply:
Yeah so this is this way cause it can be MessageWithDates or `{
  date: Message<At, Ch, Co, Ev, Me, Re, Us>['created_at'] | string;
  type: 'message.date';
}` for DateSeparator